webapp/src/containers/LinkModal/index.tsx

```diff
@@ -1,138 +1,126 @@
+        setProjectDetails({...projectDetails, organization: (e.target as HTMLInputElement).value});
+    };
 
-    const onOrganizationChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkOrganizationError: ''});
-        setState({...state, linkOrganization: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Set project name
+    const onProjectChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setErrorState({...errorState, project: ''});
+        setProjectDetails({...projectDetails, project: (e.target as HTMLInputElement).value});
+    };
 
-    const onProjectChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkProjectError: ''});
-        setState({...state, linkProject: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Handles on confirming link project
+    const onConfirm = () => {
+        const errorStateChanges: LinkPayload = {
+            organization: '',
+            project: '',
+        };
 
-    const onConfirm = useCallback(() => {
-        if (state.linkOrganization === '') {
-            setError((value) => ({...value, linkOrganizationError: 'Organization is required'}));
+        if (projectDetails.organization === '') {
+            errorStateChanges.organization = 'Organization is required';
         }
-        if (state.linkProject === '') {
-            setError((value) => ({...value, linkProjectError: 'Project is required'}));
+
+        if (projectDetails.project === '') {
+            errorStateChanges.project = 'Project is required';
         }
 
-        if (!state.linkOrganization || !state.linkProject) {
+        if (errorStateChanges.organization || errorStateChanges.project) {
+            setErrorState(errorStateChanges);
             return;
         }
 
-        // Create payload to send in the POST request.
-        const payload = {
-            organization: state.linkOrganization,
-            project: state.linkProject,
-        };
-
-        // TODO: save the payload in a state variable to use it while reading the state
-        // we can see later if there exists a better way for this
-        setLinkPayload(payload);
-
         // Make POST api request
-        usePlugin.makeApiRequest(Constants.pluginApiServiceConfigs.createLink.apiServiceName, payload);
-    }, [state]);
+        linkTask(projectDetails);
+    };
 
-    useEffect(() => {
-        if (linkPayload) {
-            const {isLoading, isSuccess, isError} = usePlugin.getApiState(Constants.pluginApiServiceConfigs.createLink.apiServiceName, linkPayload);
-            setLoading(isLoading);
-            if (isSuccess) {
-                onHide();
-            }
-            if (isError) {
-                setAPIError('Organization or project name is wrong');
-            }
+    // Make POST api request to link a project
+    const linkTask = async (payload: LinkPayload) => {
```

Comment:
See if you can use `useCallback` here.

Reply:
I think we don't need this here as we are not caching anything here and it's called only 1 time on each user interaction only and no state variables are changing so we don't have any dependency to observe the change